node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -264,13 +278,22 @@ impl GlobalStateSynchronizer {
     fn handle_fetched_trie<REv>(
         &mut self,
         trie_hash: Digest,
-        request_root_hashes: HashSet<Digest>,
         trie_accumulator_result: Result<Box<TrieRaw>, TrieAccumulatorError>,
         effect_builder: EffectBuilder<REv>,
     ) -> Effects<Event>
     where
         REv: From<TrieAccumulatorRequest> + From<ContractRuntimeRequest> + Send,
     {
+        let request_root_hashes = if let Some(hashes) = self.in_flight.remove(&trie_hash) {
+            hashes
+        } else {
+            warn!(
```

Comment:
If it's a bug, I would bump this to `error!()`

Reply:
I think the idea was for `error!` to be reserved for critical issues that need immediate attention? This one is not critical in itself, hence I thought a `warn!` makes more sense.

Reply:
If the code path is supposed to be unreachable, then it should be an `error!`. If it's not supposed to be unreachable, then it should be a `warn!`.

To me it seems like it is unreachable in the current state of the code, with the only interactions with the list being when handling a `FetchTrie` request and `handle_fetched_trie`. I would go for `error!` but I don't want to block the PR for this.

Reply:
I guess the main question here is what the semantics of `error!` vs `warn!` are. It's been suggested at some point, IIRC, that `error!` should be for wake-up-in-the-middle-of-the-night kind of errors, but I'm not sure if we're sticking to this convention. If not, then I guess it could make sense to make this an error :+1: